getDateFromBancAccaunt/account_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# Information
+# autor: Serghei Scepanovschi
+# account_spe.rb ver2.0
+#
+
+require_relative 'example_bank'
+require 'rspec'
+str = IO.read("temp.json")
+my_hash = JSON.parse(str)
+accounts = []
+my_hash['account'].each do |item|
+  accounts << Account.new(
+    item['name'],
+    item['currency'],
+    item['balance'].to_f,
+    item['nature']
+  )
+end
+example_bank = ExampleBank.new
+describe 'accounts' do
```

Comment:
Refactor spec.
You need to save 2 HTML files accounts.html and transactions.html
and cover with specs methods for parse accounts and parse transactions.
You should show in spec:
1) Count of accounts expect(accounts.count).to eq(5)
2) Count of transactions expect(transactions.count).to eq(10)
3) Example of one account in hash format (check again example)

  expect(accounts.first.to_hash).to eq(
    {

      "name"         => "Demo Everyday Account",
      "currency"     => "USD",
      "balance"      => 1959.90,
      "nature"       => "account",
      "transactions" => []
    }
  )
     4. Example of one transaction in hash format (check my example).

expect(transactions.first.to_hash).to eq(
  {
    "date":         "2020-10-07",
    "description":  "bought food",
    "amount":       -89.31,
    "currency":     "USD",
    "account_name": "Demo Everyday Account"
  }
)

Reply:
I have a lot of pages for transactions. I parsed each transaction from diferent page. Do i have to save all of them? Are you tolking about trasactios list not about pages where is trasaktion's details. My metod pars_transaction works with trasaktion's details pages is diferent that you described it doesn't work with one html page it works with many diferent pages. To test it i have to save all srucrure of the site. Why may't i compare each field of class in the spec?
What is wrond with hash?

Reply:
Could you be so kind to give me little bit more inputs?

Reply:
You could refactor your code so that it loads all transactions for the 2 months period
In this way you would use only one html page for one account, thus decreasing the number of files used and the
navigation on the interface.

In rspec we should see an example of the Account and Transaction Class Object as in the example we provided you with above.

Good luck